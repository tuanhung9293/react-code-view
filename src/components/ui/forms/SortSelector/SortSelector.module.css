.wrapper{
    margin: 5px 0;
    overflow: auto;
}

.labels{
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    margin: 5px 0;
}