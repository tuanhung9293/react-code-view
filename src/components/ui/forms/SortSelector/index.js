import React from 'react'
import Select from 'components/ui/forms/Select'
import Label from '../Label'
import {AVAILABLE_SORT_TYPES} from 'constants/defaultValues'
import styles from './SortSelector.module.css'

function SortSelector({label, ...rest}) {
  return (
    <div className={styles.wrapper}>
      <div className={styles.labels}>
        <Label>{label}</Label>
      </div>
      <Select {...rest}>
        {AVAILABLE_SORT_TYPES.map(type => (
          <option key={type.id} value={type.value}>{type.message}</option>
        ))}
      </Select>
    </div>
  )
}

export default SortSelector
