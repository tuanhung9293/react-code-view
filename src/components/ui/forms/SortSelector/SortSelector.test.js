import React from 'react'
import {shallow} from 'enzyme'
import SortSelector from './'


describe('renders', () => {
  it('renders SortSelector', () => {
    expect(shallow(
        <SortSelector />
    )).toMatchSnapshot()
  })

  it('SortSelector should pass props down', () => {
    expect(shallow(
        <SortSelector data1="data 1" data2="data 2" />
    )).toMatchSnapshot()
  })
})