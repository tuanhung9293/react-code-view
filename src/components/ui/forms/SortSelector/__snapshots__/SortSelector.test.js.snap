// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SortSelector should pass props down 1`] = `
<div
  className="wrapper"
>
  <div
    className="labels"
  >
    <Label />
  </div>
  <Select
    data1="data 1"
    data2="data 2"
  >
    <option
      key="editDec"
      value="dateEdited,1"
    >
      Default
    </option>
    <option
      key="titleDec"
      value="title,1"
    >
      Title decrease
    </option>
    <option
      key="dateDec"
      value="dateCreated,1"
    >
      Date decrease
    </option>
    <option
      key="titleInc"
      value="title,-1"
    >
      Title increase
    </option>
    <option
      key="dateInc"
      value="dateCreated,-1"
    >
      Date increase
    </option>
  </Select>
</div>
`;

exports[`renders renders SortSelector 1`] = `
<div
  className="wrapper"
>
  <div
    className="labels"
  >
    <Label />
  </div>
  <Select>
    <option
      key="editDec"
      value="dateEdited,1"
    >
      Default
    </option>
    <option
      key="titleDec"
      value="title,1"
    >
      Title decrease
    </option>
    <option
      key="dateDec"
      value="dateCreated,1"
    >
      Date decrease
    </option>
    <option
      key="titleInc"
      value="title,-1"
    >
      Title increase
    </option>
    <option
      key="dateInc"
      value="dateCreated,-1"
    >
      Date increase
    </option>
  </Select>
</div>
`;
